_posts/2016-04-26-distributed-game-of-life-in-elixir.md

```diff
@@ -0,0 +1,1065 @@
+---
+layout: post
+title:  "Distributed Game of Life in Elixir"
+date:   2016-04-26 22:34:00 +0100
+author: "Artur Trzop"
+categories: elixir
+tags: elixir distributed game
+---
+
+I wrote my first game in Elixir. It's a popular thing - Conway's Game of Life - but it gets quite interesting when you solve it in a functional language, especially when you can see how [actor model](https://en.wikipedia.org/wiki/Actor_model) works and how actors are distributed across servers in network.
+
+<img src="/images/blog/posts/distributed-game-of-life-in-elixir/game_of_life_logo.png" style="float:right;margin-left:20px;" />
+
+__What I'm going to show you is:__
+
+* How to write rules for game of life with tests in Elixir
+* Parallel tasks across lightweight processes (actors) so we utilize all CPU cores
+* How to distribute work across nodes so the game can be executed by many servers in the cluster
+* How to use GenServer behaviour, TaskSupervisor and Agents in Elixir
+
+This project and the full [source code can be found here](https://github.com/BeyondScheme/elixir-game_of_life).
```

Comment:
@Tomabana Sounds too official.

This project and the full [source code can be found on our github]